module/zfs/spa.c

```diff
@@ -7527,6 +7640,11 @@ spa_vdev_attach(spa_t *spa, uint64_t guid, nvlist_t *nvroot, int replacing,
 
 	newvd = newrootvd->vdev_child[0];
 
+	if (spa_vdev_new_spare_would_cause_double_spares(nvroot, pvd)) {
```

Comment:
We have other conditions related to spare activation around line 7602.  Wouldn't it be cleaner to add another condition there instead?

Reply:
If I moved it to 7602 then `pvd` would not be assigned yet.

Reply:
I meant old 7602, i.e. ~70 lines down, after `If the source is a hot spare, and the parent isn't already a`.

Reply:
...it still works. I've moved the check in my latest push.

Reply:
I was thinking about some code cleanup/simplification rather than just moving the function call, since some checks are already done by that point, just in somewhat different way.